### ISSUE TITLE
Steam branch selection seems to be broken

### ISSUE THREAD

**Issue by hjri** [GameServerManagers/LinuxGSM#1913]
Hello, i'm trying to run a Killing Floor 2 server with steam branch `preview`.

Here's my postdetails:
https://hastebin.com/yiyuxipoqi

Server installed initially in non-beta (`public` branch) variant. I changed `branch=""` to `branch="preview"` and tried to update - log shows that i am already at latest version. Changing it to `branch="-beta preview"` didn't help either. Game itself, running on [preview] branch tells me that server is outdated.

I only found workaround to use `branch="-beta preview"`and re-install the server via `./kf2server install`. After doing so server seems to be updated, however trying to update it via `./kf2server update` yields this in output: 
```
Update available:
        Current build: 2820476
        Available build: 2743069
        https://steamdb.info/app/232130/
```
2820476 is 'preview' build
2743069 is 'public' build
it is trying to downgrade the server to non-beta version.

I did some digging in functions and it seems that:
1. https://github.com/GameServerManagers/LinuxGSM/blob/master/lgsm/functions/update_steamcmd.sh#L162 `branchname` variable left completely unused
2. https://github.com/GameServerManagers/LinuxGSM/blob/master/lgsm/functions/update_steamcmd.sh#L169 `grep` here find first mentioning of `buildid` which happens to be `buildid` of `public` branch, the command before grep outputs a lot of info for each of branches in JSON, and branch needed buildid gets ignored.

**Comment by dgibbs64** [comment]
@hjri Thanks for this. Will need reviewing and fixing

**Comment by hjri** [comment]
Little thing i learned yesterday when switching back to "public" branch - you **have to** change it to "-beta public" to switch back, leaving it blank won't do anything - it seems that steamcmd "remembers" which branch you was last time. Nearly lost my mind over it during countless dualboots back and forth yesterday.

**Comment by ernado** [comment]
While trying to switch to Stationeers beta branch I've encountered the same issue.
Normal update process is not working, because we are always picking the first `bildid` which is `stable`, not using the provided branch info during that process.

Also the `branch` config is quite confusing, because technically it is be full flag like `-beta ${branch_name$}`, like `-beta beta` for Stationeers. 

So if anyone is trying to run Stationeers server on beta branch, use the following:
In `stserver.cfg`:
```
# Stationeers branch
branch="-beta beta"
```
To update:
```bash
$ ./stserver force-update
```

**Comment by VYRNACH** [comment]
I'm running Killing Floor 2 servers and recently opted-in to the branch="-beta preview". With the new update released, I've tried to revert this change with the following but **none of it works**:
1. (like above) switching to branch="-beta public"
2. branch="-beta "
2. branch=""
Still doesn't work even though the new update does get detected:
Update available:
        Current build: 3123684
        Available build: 3134251
        https://steamdb.info/app/232130/


I had to manually trigger the steamcmd update manually by cd to the steamcmd folder and running **app_update 232130 -beta "public"**

No update available:
        Current version: 3134251
        Available version: 3134251
        https://steamdb.info/app/232130/

Hope it helps someone

**Comment by ernado** [comment]
@VYRNACH can you please try the current `develop` LinuxGSM git branch?
This issue should be fixed in it.
(You will still need to use `branch="-beta public"` when reverting from `branch="-beta preview"`, the `branch=""` will not update the current cached branch in steamcmd)
